source/MaterialXCore/Node.cpp

```diff
@@ -647,22 +647,38 @@ void NodeGraph::setNodeDef(ConstNodeDefPtr nodeDef)
     }
 }
 
-InputPtr Node::addInputFromNodeDef(const string& name)
+InputPtr Node::addInputFromNodeDef(const string& name, NodeDefPtr fromNodeDef)
```

Comment:
I see where this additional input could be used as an optimization, allowing many calls to addInputFromNodeDef to share a single getNodeDef query, but it does add complexity to common use cases, and the alternative implementation proposed below should be even faster.